my-app/src/state/routingAtoms.ts

```diff
@@ -0,0 +1,31 @@
+import { atom } from 'jotai';
+import { api } from '../api/api';
+import type { LatLngTuple } from '../types';
+
+// State Atoms
+export const srcPosAtom = atom<LatLngTuple>([0, 0]);
+export const tgtPosAtom = atom<LatLngTuple>([0, 0]);
+export const routeAtom = atom<LatLngTuple[] | null>(null);
+export const loadingAtom = atom<boolean>(false);
+export const errorAtom = atom<string | null>(null);
+
+// Action Atom
+export const fetchRouteAtom = atom(
```

Comment:
Interesting approach, what if you had to receive another parameter for the calculation?